### ISSUE TITLE
Handle multiple relations in getEndpoint

### ISSUE THREAD

**Issue by kentcdodds** [js-data/js-data-http#33]
versions:

```
"js-data": "2.8.2",
"js-data-angular": "3.1.0",
```

Hi there! I have a model that looks like this:

```javascript
{
  name: 'TelephoneNumber',
  endpoint: 'telephonenumber',
  relations: {
    belongsTo: {
      Account: {
        parent: true,
        localKey: 'accountId',
      },
      Partition: {
        parent: true,
        localKey: 'partitionId',
      }
    }
  },
}
```

As you can see a `TelephoneNumber` belongs to both an `Account` and `Partition` (incidentally, an `Account` belongs to a `Partition`). Odd, I know. When I call `getEndpoint` like so:

```javascript
const item = {accountId: 'foo', partitionId: 'bar', ...otherStuff}
const url = httpAdapter.getEndpoint(TelephoneNumberDefinition, item)
```

Here this returns: `partition/bar/telephonenumber`

The issue is that right now the `defineResource` method doesn't support having a definition with multiple `parent`, `parentKey`, and `parentId` properties ([here](https://github.com/js-data/js-data/blob/master/src/datastore/sync_methods/defineResource.js#L123-L127). So when that runs the account is set as the parent, and then the partition overrides it (though this is not 100% deterministic because iteration on objects is not 100% consistent across browsers).

What I would expect is if I provide an `accountId` it should factor the `account` in generating the URL, so it would become: `partition/bar/account/foo/telephonenumber`

Problem I can see from this is with me providing both an `accountId` and a `partitionId`, you don't know which one to use... So, I think maybe the solution could be that I don't provide a `partitionId` and you look that up on the `account` which would happen naturally...

To sum up, what I think needs to happen is [this code](https://github.com/js-data/js-data/blob/master/src/datastore/sync_methods/defineResource.js#L123-L127) needs to account for multiple parents. Then all code using that needs to account for a potential of an array of parents (in particular, the `getEndpoint` method). That would resolve my issue...

I expect that this would be a breaking change for people depending on and using the `parent` with multiple belongsTo.

**Comment by kentcdodds** [comment]
We found a workaround that works pretty well. Technically I still think this is a bug because the implication is that you can have multiple parents but the reality is that it's not well supported. However, because we found a workaround, we no longer have an urgent use case and you can close this issue...

If anyone comes across this and is wondering what we ended up doing, we pretty much just hardcoded the URLs needed for the Partition relationship. There weren't many places we needed that anyway.

**Comment by kentcdodds** [comment]
cc @TheMcMurder. This getting fixed might help you out at some point.
